Zend/zend_inheritance.c

```diff
@@ -619,9 +619,7 @@ static void do_inheritance_check_on_method(zend_function *child, zend_function *
 		zend_string *method_prototype = zend_get_function_declaration(parent);
 		zend_string *child_prototype = zend_get_function_declaration(child);
 
-		if (child->common.prototype && (
-			child->common.prototype->common.fn_flags & ZEND_ACC_ABSTRACT
-		)) {
+		if (parent->common.fn_flags & ZEND_ACC_ABSTRACT) {
```

Comment:
This doesn't look quite right to me. Doesn't think mean that if B extends A implements I, and B violates I (and not just A), then you will only get a warning, rather than a fatal error?